### PR TITLE
fix(menu): HTMLelement anchor support and fix on first time added (VIV-1245) (VIV-1246)

### PR DESCRIPTION
**Pull request by rinaok**

### CODE REVIEW

**Review by rachelbt — COMMENTED** [review]
didn't looked at all the code - but if the button + menu is inside dlex wrapper - the menu is badly calculated:
image
```

<div style="text-align: end; display: flex; justify-content: flex-end">
  <vwc-button id="button" label="Toggle Menu" onclick="menu.open = !menu.open" appearance="outlined" ></vwc-button>

  <vwc-menu id="menu" anchor="button" placement="right-start" open>
    <vwc-menu-item text="Menu item 1"></vwc-menu-item>
    <vwc-menu-item text="Menu item 2"></vwc-menu-item>
  </vwc-menu>
</div>

```

File: libs/components/src/lib/menu/menu.template.ts [hunk 1]
```diff
@@ -44,7 +43,6 @@ export const MenuTemplate: (
 				:anchor=${(x) => x.anchor}
 				@vwc-popup:open="${x => handlePopupEvents(x, true)}"
 				@vwc-popup:close="${x => handlePopupEvents(x, false)}"
-				${ref('_popup')}
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.template.ts:47]: why is this not needed?
Select + Combo has it.

Comment (rinaok) [reply libs/components/src/lib/menu/menu.template.ts:47]: We can remove it there as well, we use the _popup only for autoDismiss, and I updated it to work like the tooltip and toggleitp.

**Comment by rinaok** [comment]
> didn't looked at all the code - but if the button + menu is inside dlex wrapper - the menu is badly calculated: image
> 
> ```
> 
> <div style="text-align: end; display: flex; justify-content: flex-end">
>   <vwc-button id="button" label="Toggle Menu" onclick="menu.open = !menu.open" appearance="outlined" ></vwc-button>
> 
>   <vwc-menu id="menu" anchor="button" placement="right-start" open>
>     <vwc-menu-item text="Menu item 1"></vwc-menu-item>
>     <vwc-menu-item text="Menu item 2"></vwc-menu-item>
>   </vwc-menu>
> </div>
> ```

Did it act differently before my changes? Can you put a screenshot of how it worked before?

File: libs/components/src/lib/menu/menu.ts [hunk 2]
```diff
@@ -52,34 +61,58 @@ export class Menu extends FastMenu {
...
+		document.removeEventListener('click', this.#closeOnClickOutside);
+	}
 
-		if (this.autoDismiss) {
-			document.removeEventListener('click', this.#dismissOnClickOutside);
-		}
+	#setupAnchor(a: HTMLElement) {
+		a.addEventListener('click', this.#openIfClosed, true);
+		a.setAttribute('aria-haspopup', 'menu');
+		// TODO aria-controls="myid"
```
Comment (YonatanKra) [inline libs/components/src/lib/menu/menu.ts:98]: What is that? Should it be a JIRA issue?

**Comment by rinaok** [comment]
> didn't looked at all the code - but if the button + menu is inside dlex wrapper - the menu is badly calculated: image
> 
> ```
> 
> <div style="text-align: end; display: flex; justify-content: flex-end">
>   <vwc-button id="button" label="Toggle Menu" onclick="menu.open = !menu.open" appearance="outlined" ></vwc-button>
> 
>   <vwc-menu id="menu" anchor="button" placement="right-start" open>
>     <vwc-menu-item text="Menu item 1"></vwc-menu-item>
>     <vwc-menu-item text="Menu item 2"></vwc-menu-item>
>   </vwc-menu>
> </div>
> ```

@rachelbt https://vivid-a53dpv0c6ry0.deno.dev/components/menu#placement

File: libs/components/src/lib/menu/README.md [hunk 3]
```diff
@@ -35,27 +35,36 @@ _Menu_ internally uses _popup_ to display an element and its descendants above t
...
 
-_Menu_ internally uses _popup_ to display an element and its descendants above the rest of the document.
+Use the `anchor` attribute to link the menu to the element responsible for showing and hiding it. It can be the ID or a reference to said element.
 
-`anchor` property from _popup_ propagate through _menu_ and sets its anchor reference.
+The anchor must be clickable and in most cases, will be a button displaying an information glyph as in the example below.
 
-- Type: `string`
-- Default: `''`
+- Type: `string | HTMLElement`
```
Comment (YonatanKra) [inline libs/components/src/lib/menu/README.md:42]: Do we really want to support this for a menu? IMO, as opposed to `tooltip` or `toggletip`, a menu is not something that's added in vast quantities.

File: libs/components/src/lib/menu/menu.template.ts (hunk 1 above)
Comment (YonatanKra) [reply libs/components/src/lib/menu/menu.template.ts:47]: Can you create an issue to our backlog to change `select` and `combo` too, please?

File: libs/components/src/lib/menu/menu.ts (hunk 2 above)
Comment (YonatanKra) [reply libs/components/src/lib/menu/menu.ts:98]: None :(
Who knows what this is?
I just created an issue to research this. In the backlog. Whenever someone feels like it.
https://jira.vonage.com/browse/VIV-1254

**Review by YonatanKra — APPROVED** [review]
LGTM. Notice visual regression fails, though...

**Pull request merged**